scripts/build-windows-provisioning-scripts.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash -e
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+
+required_env_vars=(
+    "CLIENT_ID"
+    "CLIENT_SECRET"
+    "RESOURCE_GROUP_NAME"
+    "STORAGE_ACCOUNT_NAME"
+    "STORAGE_CONTAINER_NAME"
+    "SUBSCRIPTION_ID"
+    "TENANT_ID"
+)
+
+for v in "${required_env_vars[@]}"
+do
+    if [ -z "${!v}" ]; then
+        echo "$v was not set!"
```

Comment:
Maybe create a list of all the unset env vars and exit 1 after so you don't have to run several times

Reply:
I believe the way to accomplish this is using `set -o nounset` and 

```
${FOO:?Please provide a value for FOO}
```

Reply:
Updated to create a list of all unset env vars following @ksubrmnn's suggestion.